File: pkg/runtime/javascript/fixtures/transformer/parameters.out.airplane.js
import airplane from "airplane";

export default airplane.task(
  {
    slug: "my_task",
    parameters: {
      all: {
        name: "All fields",
        description: "My description",
        type: "shorttext",
        required: false,
        default: "My default",
        regex: "^.*$",
        options: [{
          label: "Thing 3",
          value: "Secret gremlin"
        }]
      },
      shorttext: {
        type: "shorttext",
        default: "Text"
      },
      longtext: {
        type: "longtext",
        default: "Longer text"
      },
      sql: {
        type: "sql",
        default: "SELECT 1"
      },
      boolean_true: {
        type: "boolean",
        default: true
      },
      boolean_false: {
        type: "boolean",
        default: false
      },
      upload: {
        type: "upload",
        default: "upl123"
      },
      integer: {
        type: "integer",
        default: 10
      },
      integer_zero: {
        type: "integer",
        default: 0
      },
      float: {
        type: "float",
        default: 3.14
      },
      float_zero: {
        type: "float",
        default: 0
      },
      date: {
        type: "date",
        default: "2006-01-02"
      },
      datetime: {
        type: "datetime",
        default: new Date("2006-01-02T15:04:05Z07:00")
      },
      configvar: {
        type: "configvar",
        default: "MY_CONFIG"
      },
      configvar_legacy: {
        type: "configvar",
        default: "MY_CONFIG"
      }
    }
  },
  async () => {
    return []
  },
);

File: pkg/definitions/updaters/python/fixtures/parameters_out_airplane.py
from datetime import date, datetime, timezone
from typing import Optional
from typing_extensions import Annotated
import airplane


@airplane.task()
def my_task(
    simple: str,
    default_name: str,
    all: Annotated[
        Optional[str],
        airplane.ParamConfig(
            name="All fields",
            description="My description",
            regex="^.*$",
            options=[
                "Thing 1",
                "Thing 2",
                airplane.LabeledOption(label="Thing 3", value="Secret gremlin"),
            ],
        ),
    ] = "My default",
    shorttext: str = "Text",
    longtext: airplane.LongText = "Longer text",
    sql: airplane.SQL = "SELECT 1",
    boolean_true: bool = True,
    boolean_false: bool = False,
    upload: airplane.File = "upl123",
    integer: int = 10,
    integer_zero: int = 0,
    float: float = 3.14,
    float_zero: float = 0,
    date: date = date(2006, 1, 2),
    datetime: datetime = datetime(2006, 1, 2, 15, 4, 5, tzinfo=timezone.utc),
    datetime2: datetime = datetime(2006, 1, 2, 15, 4, 5, 123000, tzinfo=timezone.utc),
    configvar: airplane.ConfigVar = airplane.ConfigVar("MY_CONFIG"),
    configvar_legacy: airplane.ConfigVar = airplane.ConfigVar("MY_CONFIG"),
    default_name_optional: Optional[str] = None,
):
    pass
